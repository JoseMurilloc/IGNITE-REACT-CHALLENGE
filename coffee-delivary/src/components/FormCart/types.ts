export type MethodsPayment = 'credit' | 'debit' | 'money'
export type ListPayments = Array<{ type: MethodsPayment; label: string }>
