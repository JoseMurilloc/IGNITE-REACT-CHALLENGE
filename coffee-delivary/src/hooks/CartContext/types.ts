import React from 'react'
import { CoffeeDTO } from '../../dtos/Coffee'

export interface CartContextData {
  coffees: CoffeeDTO[]
  addCoffeeCart: (coffee: CoffeeDTO) => void
  removeCoffeeCart: (idCoffee: number) => void
  incrementCoffeeCart: (idCoffee: number) => void
  decrementCoffeeCart: (idCoffee: number) => void

  resetCart: () => void
}
export type AddCoffeeCartParams = {
  coffee: CoffeeDTO
}
export type CartContextProviderProps = {
  children: React.ReactNode
}

export type Cart = {
  coffees: CoffeeDTO[]
}
