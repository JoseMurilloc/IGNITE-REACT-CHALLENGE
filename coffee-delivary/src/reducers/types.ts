export interface Coffee {
  types: string[]
  title: string;
  description: string;
  price: number; 
  image: string;
}

export interface CartState {
  coffees: Coffee[]
}
