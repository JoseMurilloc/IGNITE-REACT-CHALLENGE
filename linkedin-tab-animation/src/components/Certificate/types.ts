export type CertificateProps = {
	title: string;
	educationalInstitution: string;
	dateEmit: string;
	image: string;
};
