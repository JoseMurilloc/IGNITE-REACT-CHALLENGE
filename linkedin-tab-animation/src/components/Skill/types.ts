export type SkillProps = {
	name: string;
};
