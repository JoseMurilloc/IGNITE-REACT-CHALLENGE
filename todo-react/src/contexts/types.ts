import React from 'react'
import { Task } from '../reducer/tasks/types'

export interface TaskContextData {
  tasks: Task[]
  dispatch: any
}

export type TaskContextProviderProps = {
  children: React.ReactNode
}
