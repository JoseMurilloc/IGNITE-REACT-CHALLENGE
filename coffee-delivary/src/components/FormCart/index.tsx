import { CurrencyDollar, MapPinLine, Money } from 'phosphor-react';
import { useState } from 'react';
import { useFormContext } from 'react-hook-form';
import { Input } from '../Input';
import { SelectControl } from '../SelectControl';
import { SelectPaymentButton } from '../SelectPaymentButton';
import styles from './styles.module.css';
import { ListPayments, MethodsPayment } from './types';

const METHOD_PAYMENTS: ListPayments = [
  {type: 'credit', label: 'Cartão de Crédito'},
  {type: 'debit', label: 'Cartão de Débito'},
  {type: 'money', label: 'Dinheiro'},
]

export function FormCart () {
  const [selectMethodPayment, setSelectMethodPayment] = useState<MethodsPayment>('credit')
  const {setValue} = useFormContext()

  const handleSelectTypePayment = (type: MethodsPayment) => {
      setValue('paymentType', type)
      setSelectMethodPayment(type)
  }


  return (
    <>  
      <section className={styles.contentInfoOfPurchase}>
        <header>
          <MapPinLine size="2.2rem" color="#C47F17" />
          <div className={styles.contentHeader}>
            <span>Endereço de Entrega</span>
            <span>Informe o endereço onde deseja receber seu pedido</span>
          </div>
        </header>
        <div className={styles.containerInputs}>
          <div style={{width: 200}}>
            <Input 
              registerName="cep" 
              type="number" 
              placeholder="CEP" 
            />
          </div>
          <div>
            <Input 
              registerName="street" 
              type="text" 
              placeholder="Rua" 
            />
          </div>
          <div>
            <Input 
              registerName="number" 
              type="text" 
              placeholder="Número" 
            />
            <Input 
              registerName="complement" 
              type="text" 
              placeholder="Complemento" 
            />
          </div>
          <div>
            <Input 
              registerName="district" 
              type="text" 
              placeholder="Bairro" 
            />
            <Input 
              registerName="city" 
              type="text" 
              placeholder="Cidade" 
            />
            <SelectControl registerName="uf" />
          </div>
        </div>
      </section>
      <section className={styles.formPaymentContainer}>
        <header>
          <CurrencyDollar size="2.2rem" color="#8047F8" />
          <div className={styles.contentHeader}>
            <span>Pagamento</span>
            <span>
              O pagamento é feito na entrega. Escolha a forma que deseja pagar
            </span>
          </div>
        </header>
        <aside className={styles.paymentsContainer}>
          {METHOD_PAYMENTS.map(({type, label}) => (
            <SelectPaymentButton  
              onClick={() => handleSelectTypePayment(type)}
              typePayment={type} 
              label={label} 
              selected={selectMethodPayment === type}
            />
          ))}
        </aside>
      </section>
    </>
  );
}
