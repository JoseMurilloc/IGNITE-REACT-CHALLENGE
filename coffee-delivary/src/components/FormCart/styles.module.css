.contentInfoOfPurchase {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding: 4rem;
  gap: 3.2rem;

  height: 37.2rem;
  background: #F3F2F2;
  border-radius: 0.6rem;
}

.contentInfoOfPurchase header {
  display: flex;
  gap: 0.4rem;
}


header .contentHeader {
  display: flex;
  flex-direction: column;
}

header .contentHeader span {
  font-family: 'Roboto';
  font-size: 1.4rem;
  line-height: 130%;
  color: #403937;
}

header .contentHeader span + span {
  color: #574F4D;
}

.containerInputs {
  display: flex;
  flex-direction: column;
  gap: 1.6rem;
}

.containerInputs div {
  display: flex;
  gap: 1.2rem;
}

.paymentsContainer {
  display: flex;
  gap: 1.2rem;
}


.formPaymentContainer {
  padding: 4rem;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  gap: 3.2rem;
  background: #F3F2F2;
  border-radius: 0.6rem;
}

.formPaymentContainer header {
  display: flex;
}


