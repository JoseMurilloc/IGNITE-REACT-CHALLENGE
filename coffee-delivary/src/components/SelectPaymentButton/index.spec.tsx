import { render, screen } from '@testing-library/react'
import { SelectPaymentButton } from '.'
import { MethodsPayment } from '../FormCart/types'

describe('<SelectPaymentButton />', () => {
  it('Render component correctly', () => {
    const props = {
      label: 'Cartão de crédito',
      typePayment: 'credit' as MethodsPayment,
      selected: false
    }
    render(<SelectPaymentButton {...props} />)

    expect(screen.getByText(props.label)).toBeInTheDocument()
    expect(screen.getByTestId('credit-icon')).toBeInTheDocument()
  })

  it('Should be have that render buttons styles to state selected correctly', () => {
    const props = {
      label: 'Cartão de crédito',
      typePayment: 'credit' as MethodsPayment,
      selected: false
    }
    const { container } = render(<SelectPaymentButton {...props} />)

    expect(container.parentElement).toMatchSnapshot()
  })

  it('Should be have that render buttons styles to state selected correctly', () => {
    const props = {
      label: 'Cartão de debito',
      typePayment: 'debit' as MethodsPayment,
      selected: true
    }
   const { container } = render(<SelectPaymentButton {...props} />)

   expect(container.parentElement).toMatchSnapshot()

  })
})
