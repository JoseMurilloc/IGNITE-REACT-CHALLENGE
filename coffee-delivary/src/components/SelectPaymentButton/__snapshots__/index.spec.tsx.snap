// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectPaymentButton /> Should be have that render buttons styles to state selected correctly 1`] = `
<body>
  <div>
    <button
      class="containerSelectPaymentButton"
      type="button"
    >
      <svg
        data-testid="credit-icon"
        fill="#8047F8"
        height="2rem"
        viewBox="0 0 256 256"
        width="2rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="256"
          width="256"
        />
        <rect
          fill="none"
          height="144"
          rx="8"
          stroke="#8047F8"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          width="208"
          x="24"
          y="56"
        />
        <line
          fill="none"
          stroke="#8047F8"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="168"
          x2="200"
          y1="168"
          y2="168"
        />
        <line
          fill="none"
          stroke="#8047F8"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="120"
          x2="136"
          y1="168"
          y2="168"
        />
        <line
          fill="none"
          stroke="#8047F8"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="24"
          x2="232"
          y1="96.9"
          y2="96.9"
        />
      </svg>
      <span>
        Cartão de crédito
      </span>
    </button>
  </div>
</body>
`;

exports[`<SelectPaymentButton /> Should be have that render buttons styles to state selected correctly 2`] = `
<body>
  <div>
    <button
      class="containerSelectPaymentButtonSelected"
      type="button"
    >
      <svg
        fill="#8047F8"
        height="2rem"
        viewBox="0 0 256 256"
        width="2rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="256"
          width="256"
        />
        <polygon
          fill="none"
          points="24 96 232 96 128 32 24 96"
          stroke="#8047F8"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
        />
        <line
          fill="none"
          stroke="#8047F8"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="56"
          x2="56"
          y1="96"
          y2="176"
        />
        <line
          fill="none"
          stroke="#8047F8"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="104"
          x2="104"
          y1="96"
          y2="176"
        />
        <line
          fill="none"
          stroke="#8047F8"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="152"
          x2="152"
          y1="96"
          y2="176"
        />
        <line
          fill="none"
          stroke="#8047F8"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="200"
          x2="200"
          y1="96"
          y2="176"
        />
        <line
          fill="none"
          stroke="#8047F8"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="32"
          x2="224"
          y1="176"
          y2="176"
        />
        <line
          fill="none"
          stroke="#8047F8"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="16"
          x2="240"
          y1="208"
          y2="208"
        />
      </svg>
      <span>
        Cartão de debito
      </span>
    </button>
  </div>
</body>
`;
